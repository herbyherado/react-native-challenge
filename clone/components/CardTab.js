import React, { Component } from 'react'
import { Text, View, TouchableHighlight, StyleSheet } from 'react-native' 

class CardTab extends Component {
  render() {
    return (
      <TouchableHighlight onPress = { this.props.press }>
        <View style={styles.card}> 
          <Text style={styles.cardText} > { this.props.text }</Text>
        </View>
      </TouchableHighlight>  
    )
  }
};

const styles = StyleSheet.create({
  card: {
    marginTop: 20,
    alignItems: 'center',
    alignSelf: 'center',
    padding: 30,
    width: '95%',
    borderWidth: 1,
    borderBottomColor: 'silver',
    borderTopColor: 'silver'
  },
  cardText: {
    color: '#fff',
    fontSize: 30,
    backgroundColor: 'transparent'
  },
})

export default CardTab