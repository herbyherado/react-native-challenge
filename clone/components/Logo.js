import React, { Component } from 'react';
import { View, StyleSheet, Image } from 'react-native'

class Logo extends Component {
  render() {
    return (
    <View style={styles.imgContain}>
      <Image style={ styles.image } source={require('./assets/UFC_Logo.png')}/> 
    </View>
    )
  }
};

const styles = StyleSheet.create({
  image: {
    flex: 1,
    resizeMode: 'contain',
    width: '90%',
    alignSelf: 'center',
  },
  imgContain: {
    height: 150,
    marginBottom: 20
  }
})

export default Logo