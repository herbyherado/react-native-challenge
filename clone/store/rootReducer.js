import { combineReducers } from 'redux'
import Fighter from './fighters/fighter.reducer'
import Octagirl from './octagirls/octagirls.reducer'

const reducers = combineReducers({
  fighters: Fighter,
  octagirls: Octagirl
})

export default reducers