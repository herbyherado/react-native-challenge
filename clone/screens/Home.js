import React, { Component } from 'react';
import { View, Text, TouchableHighlight, StyleSheet, Image, ImageBackground } from 'react-native'
import { StackNavigator } from 'react-navigation'
import { connect } from 'react-redux'
import { getFighters } from '../store/fighters/fighter.actions'
import { getOctagirls } from '../store/octagirls/octagirls.actions'
import CardTab from '../components/CardTab'
import Logo from '../components/Logo'

class Home extends Component {
  render() {
    return (
      <View style={styles.wrapper}>
        <Logo />
        <CardTab press={ () => this.props.navigation.navigate('Fighters') } text={'Fighters'} />
        <CardTab press={ () => this.props.navigation.navigate('Octagirls') } text={'Octagon Girls'} />
      </View>
    )
  }
  componentDidMount () {
    this.props.loadFighters()
    this.props.loadOctagirls()
  }
};

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    backgroundColor: '#303030',
    width: '100%',
    paddingTop: 20
  }
})

function mapDispatchToProps (dispatch) {
  return {
    loadFighters: () => dispatch(getFighters()),
    loadOctagirls: () => dispatch(getOctagirls())
  }
}

export default connect(null, mapDispatchToProps)(Home)
