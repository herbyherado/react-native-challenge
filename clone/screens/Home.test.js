import React, { Component } from 'react'
import enzyme, { shallow, mount } from 'enzyme'
import { Text, View, Image } from 'react-native'
import Adapter from 'enzyme-adapter-react-16'
import Home from './Home'
import renderer from 'react-test-renderer'
import Logo from '../components/Logo'
import CardTab from '../components/CardTab'
import store from '../store/store'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { searchFighters } from '../store/fighters/fighter.actions'

const mockStore = configureStore()
const initialState = {
  fighters: [],
  query: ''
}
enzyme.configure({ adapter: new Adapter()})

let mstore
let wrapper

beforeEach(() => {
  mstore = mockStore(initialState)
  wrapper = shallow(<Provider store={store}><Home/></Provider>)
})

describe('Snapshot test <Home/>',()=>{
  it('capturing Snapshot of Home', () => {
      const renderedValue =  renderer.create(<Provider store={store}><Home/></Provider>).toJSON()
      expect(renderedValue).toMatchSnapshot();
  })
})

describe('<Home/>', () => {
  it('should render <Home/>', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper.length).toEqual(1)
  })
  it('should have components', () => {
    expect(wrapper.find(<View/>)).toBeDefined()
    expect(wrapper.find('Logo')).toBeDefined()
    expect(wrapper.find('CardTab')).toBeDefined()
  })
})

describe('Redux check <Home/>', () => {
  it('check props for <Home/>', () => {
    expect(wrapper.prop('output')).toEqual(initialState.output)
  })
})

describe('<CardTab/> should be pressable', () => {
  it('tab should be pressable', () => {
    let mockCallBack = jest.fn()
    let wrap = shallow(<CardTab press={ () => mockCallBack } text={'Fighters'} />)
    wrap.find('TouchableHighlight').props().onPress()
    expect(mockCallBack.mock.calls.length).toEqual(0)
    expect(wrap.props()).toHaveProperty('onPress')
  })
})

describe('<Home/> should render instance', () => {
  it('should render navigate instances', () => {
    let inst = wrapper.instance()
    expect(inst.props).toBeTruthy()
  })
})
