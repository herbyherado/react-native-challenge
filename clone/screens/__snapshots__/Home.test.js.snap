// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test <Home/> capturing Snapshot of Home 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#303030",
      "flex": 1,
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "height": 150,
        "marginBottom": 20,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../components/assets/UFC_Logo.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "flex": 1,
          "resizeMode": "contain",
          "width": "90%",
        }
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "borderBottomColor": "silver",
          "borderTopColor": "silver",
          "borderWidth": 1,
          "marginTop": 20,
          "padding": 30,
          "width": "95%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontSize": 30,
          }
        }
      >
         
        Fighters
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "borderBottomColor": "silver",
          "borderTopColor": "silver",
          "borderWidth": 1,
          "marginTop": 20,
          "padding": 30,
          "width": "95%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontSize": 30,
          }
        }
      >
         
        Octagon Girls
      </Text>
    </View>
  </View>
</View>
`;
